tensorboard/webapp/metrics/views/card_renderer/image_card_component.ts

```diff
@@ -73,6 +77,94 @@ export class ImageCardComponent {
     this.stepIndexChange.emit($event.value as number);
   }
 
+  changeDistinct(change: SimpleChange) {
+    return change.currentValue !== change.previousValue;
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (
+      (changes['selectedSteps'] &&
+        this.changeDistinct(changes['selectedSteps'])) ||
+      (changes['selectedTime'] && this.changeDistinct(changes['selectedTime']))
+    ) {
+      this.renderRangeSlider();
+    }
+  }
+
+  renderRangeSlider() {
+    if (
+      !this.selectedTime ||
+      !this.selectedTime.endStep ||
+      this.selectedTime.clipped
+    ) {
+      return;
+    }
+
+    const boundSize = this.stepValues.length - 1;
+    const sliderUnit = 1 / boundSize;
+    const startStep =
+      this.selectedTime.startStep < this.stepValues[0]
+        ? this.stepValues[0]
+        : this.selectedTime.startStep;
+    const endStep =
+      this.selectedTime.endStep > this.stepValues[boundSize]
+        ? this.stepValues[boundSize]
+        : this.selectedTime.endStep;
+    let startPosition = 0;
+    let width = 0;
+
+    let i = 0;
+    for (; i < this.stepValues.length - 1; i++) {
```

Comment:
As discussed in person: I would like to see this logic encapsulated in a GetStartPosition function for readability.  I see that you reuse the i variable so maybe encapsulate it all in a GetStartPositionAndWidth function or just recalculate the i value in the GetSliderWidth function.

Reply:
Thanks! It's a good suggestion. I prefer not recalculate the width so I put it as `getStartPositionAndWidth()` and add comments before each for loop chunk. PTAL.